modules/internal/ExternalArray.chpl

```diff
@@ -55,7 +55,8 @@ module ExternalArray {
         halt("there should only be one set of indices, not multiple dimensions");
       }
       var r = inds(1);
-      if (r.low != 0) {
+      // Allow empty arrays (which have the domain 1..0)
```

Comment:
0..-1 is just as reasonable for an empty array.
It seems to me that if you want to check this, you'd check for r.low < r.high.

Reply:
I'm not personally sold that these need to support 1..0 bounds, but I'm also OK with it.

Reply:
We were running into this problem because `0..#0` was getting transformed into `1..0` (https://github.com/chapel-lang/chapel/issues/5165 tracks the desire to fix that issue).  I was using `0..#0` to create the array when we know the size, and it is a lot more work for a user to create an external array with their own domain (though if they do, {0..-1} should be accepted without additional work right now).

I don't feel strongly about whether `2..1` or any other empty domain gets supported, but I think I'd rather wait until there was a strong desire to do so.